src/lib/transactions.ts

```diff
@@ -87,10 +93,37 @@ const processTransaction = (transaction: GQLEdgeInterface, type: string) => ({
   date: ""
 });
 
+export async function checkTransactionError(
+  transaction: GQLEdgeInterface | Transaction
+): Promise<boolean> {
+  if (transaction.node.owner.address !== AGENT_TOKEN_ADDRESS) {
+    return false;
+  }
+
+  return retryWithDelay(async (attempt) => {
+    const data = await gql(
+      AF_ERROR_QUERY,
+      { messageId: transaction.node.id },
+      txHistoryGateways[attempt % txHistoryGateways.length]
+    );
+
+    if (data?.data === null && (data as any)?.errors?.length > 0) {
+      throw new Error((data as any)?.errors?.[0]?.message || "GraphQL Error");
+    }
+
+    return data.data.transactions.edges.length > 0;
+  }, 2);
```

Comment:
I think we should attach `.catch(() => false)` to `retryWithDelay` so that in case of gql error even with retries as well we return a boolean not an error.